src/core/type1_parser.js

```diff
@@ -457,10 +457,14 @@ var Type1Parser = (function Type1ParserClosure() {
     if (encrypted) {
       var data = stream.getBytes();
       var isBinary = !(
-        isHexDigit(data[0]) &&
+        (isHexDigit(data[0]) || isWhiteSpace(data[0])) &&
         isHexDigit(data[1]) &&
         isHexDigit(data[2]) &&
-        isHexDigit(data[3])
+        isHexDigit(data[3]) &&
+        isHexDigit(data[4]) &&
```

Comment:
In the specification I find the case that the first ciphertext byte may not be whitespace, but I cannot find directly why we would need to check the first eight ciphertext bytes for ASCII hexadecimal character codes. It states that "the  first 4 ciphertext bytes must obey certain restrictions", so no guarantees are given about the remaining bytes as far as I can tell, but I'm probably missing something here...

Reply:
Note (emphasis mine):
> To distinguish between binary and ASCII hexadecimal input,
the first 4 ciphertext bytes must obey certain restrictions.
**Remember that two ASCII hexadecimal characters represent 1
ciphertext byte, while 1 binary byte represents 1 ciphertext
byte.**

and furthermore:
> **At least one of the first 4 ciphertext bytes must not be one of
the ASCII hexadecimal character codes (a code for 0-9, A-F, or
a-f).** These restrictions can be satisfied by adjusting the initial
random plaintext bytes as necessary.

and finally:
> If the eexec-encrypted text is supplied in binary form, then every
byte is considered part of the ciphertext. If the eexec-encrypted
text is supplied in ASCII hexadecimal form, then ASCII white
space characters (blank, tab, carriage return and line feed) may be
freely interspersed within the encrypted text, **except in the first
eight characters.**

Hence the first 8 bytes of raw data should be the first 4 ciphertext bytes, for ASCII encoded data, as far as I'm reading it at least.

Reply:
Your last quote mentions eight ASCII characters and your first quote states that two ASCII characters are represented by one ciphertext byte, so eight ASCII characters correspond to four ciphertext bytes, which would match the remaining text.

Out of curiosity I looked at how for example PDFBox does it and they also seem to check only the first four bytes; see https://github.com/apache/pdfbox/blob/8b053ccddeb3a2722716085cf80ec7fdec43943f/fontbox/src/main/java/org/apache/fontbox/type1/Type1Parser.java#L881-L901. Interestingly, they check for whitespace in all four bytes, and that is also something I can read from your last quote: in the first eight characters (two characters per byte, so four bytes) no whitespace characters may exist.

Therefore, I *think* that the PDFBox implementation matches the specification most closely, but I think the specification is ambiguous here, so I do see how this can be interpreted in multiple ways :(

Reply:
> Out of curiosity I looked at how for example PDFBox does it and they also seem to check only the first four bytes

That does seem a bit odd to me, given the actual specification text...

> Interestingly, they check for whitespace in all four bytes,

That doesn't really seem to follow the specification as far as I can tell, given that it explicitly states:
> [...] then ASCII white space characters (...) may be freely interspersed within the encrypted text, **except** in the first eight characters.

All-in-all, given that the PR fixes an existing bug without any regressions elsewhere, is this good to go as is?

Reply:
> All-in-all, given that the PR fixes an existing bug without any regressions elsewhere

Well, I can't argue with that, and given the ambiguity I can't prove my view is correct either, so let's just go for it :-)